From 25e029370d5e0629524d565a991b6b02ccbf7945 Mon Sep 17 00:00:00 2001
From: qwang2 <quanyang.wang@windriver.com>
Date: Sun, 1 Jul 2018 08:20:35 +0800
Subject: [PATCH 3/3] ARM: dts: imx6sx: add display power domain support

Add display power domain in PGC. Use it to substitute the obsolete "gpc 2".

Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
---
 arch/arm/boot/dts/imx6sx.dtsi |   18 ++++++++++++------
 1 file changed, 12 insertions(+), 6 deletions(-)

diff --git a/arch/arm/boot/dts/imx6sx.dtsi b/arch/arm/boot/dts/imx6sx.dtsi
index 7387318..d0de399 100644
--- a/arch/arm/boot/dts/imx6sx.dtsi
+++ b/arch/arm/boot/dts/imx6sx.dtsi
@@ -896,6 +896,12 @@
 						clocks = <&clks IMX6SX_CLK_GPU>;
 					};
 
+					pd_display: power-domain@2 {
+						reg = <2>;
+						#power-domain-cells = <0>;
+						power-supply = <&reg_soc>;
+					};
+
 					pd_pci: power-domain@3 {
 						reg = <3>;
 						#power-domain-cells = <0>;
@@ -1346,7 +1352,7 @@
 						 <&clks IMX6SX_CLK_CSI>,
 						 <&clks IMX6SX_CLK_DCIC1>;
 					clock-names = "disp-axi", "csi_mclk", "disp_dcic";
-					power-domains = <&gpc 2>;
+					power-domains = <&pd_display>;
 					status = "disabled";
 				};
 
@@ -1379,7 +1385,7 @@
 					clocks = <&clks IMX6SX_CLK_PXP_AXI>,
 						 <&clks IMX6SX_CLK_DISPLAY_AXI>;
 					clock-names = "pxp-axi", "disp-axi";
-					power-domains = <&gpc 2>;
+					power-domains = <&pd_display>;
 					status = "disabled";
 				};
 
@@ -1391,7 +1397,7 @@
 						 <&clks IMX6SX_CLK_CSI>,
 						 <&clks IMX6SX_CLK_DCIC2>;
 					clock-names = "disp-axi", "csi_mclk", "disp_dcic";
-					power-domains = <&gpc 2>;
+					power-domains = <&pd_display>;
 					status = "disabled";
 				};
 
@@ -1403,7 +1409,7 @@
 						 <&clks IMX6SX_CLK_LCDIF_APB>,
 						 <&clks IMX6SX_CLK_DISPLAY_AXI>;
 					clock-names = "pix", "axi", "disp_axi";
-					power-domains = <&gpc 2>;
+					power-domains = <&pd_display>;
 					status = "disabled";
 				};
 
@@ -1415,7 +1421,7 @@
 						 <&clks IMX6SX_CLK_LCDIF_APB>,
 						 <&clks IMX6SX_CLK_DISPLAY_AXI>;
 					clock-names = "pix", "axi", "disp_axi";
-					power-domains = <&gpc 2>;
+					power-domains = <&pd_display>;
 					status = "disabled";
 				};
 
@@ -1426,7 +1432,7 @@
 					clocks = <&clks IMX6SX_CLK_VADC>,
 						 <&clks IMX6SX_CLK_CSI>;
 					clock-names = "vadc", "csi";
-					power-domains = <&gpc 2>;
+					power-domains = <&pd_display>;
 					gpr = <&gpr>;
 					status = "disabled";
 				};
-- 
1.7.9.5

